pkg/api/types.go

```diff
@@ -44,6 +44,7 @@ type Properties struct {
 	AgentPoolProfiles       []AgentPoolProfile      `json:"agentPoolProfiles"`
 	LinuxProfile            LinuxProfile            `json:"linuxProfile"`
 	WindowsProfile          WindowsProfile          `json:"windowsProfile"`
+	ExtensionsProfile       []ExtensionProfile      `json:"extensionsProfile"`
```

Comment:
However, I do recognize that the extension author can base their extension code on any parameter in the template and choose to noop on other pools. This certainly causes more confusion when looking that the API model.